src/app/auth/login/login.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit } from '@angular/core';
+import { AbstractControl, UntypedFormControl, UntypedFormGroup, ValidationErrors, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { AuthServiceService } from 'src/app/common/services/auth/auth-service.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+  loginForm !: UntypedFormGroup;
+  constructor(private _router: Router, private _authService: AuthServiceService, private _snackbar: MatSnackBar) {
+
+    this.loginForm = new UntypedFormGroup({
+      email: new UntypedFormControl('', [Validators.required, Validators.email]),
+      password: new UntypedFormControl('', [Validators.required])
+    }, [this.userTypeValidator]);
+  }
+
+  ngOnInit() {
+
```

Comment:
if we are not writing anything on ngOnInit() there is no point implementing OnInit .